spec/abci/abci++_app_requirements.md

```diff
@@ -808,67 +808,61 @@ implementation of
 
 ### Crash Recovery
 
+CometBFT and the application are expected to crash together and there should not 
+exist a scenario where the application has persisted state of a height greater than the
+latest height persisted by CometBFT.
+
+In practice, persisting the state of a height consists of three steps, the last of which 
+is the call to the application's `Commit` method, where the application is expected to
+persist/commit its state.
 On startup, CometBFT calls the `Info` method on the Info Connection to get the latest
 committed state of the app. The app MUST return information consistent with the
-last block it successfully completed Commit for.
-
-If the app successfully committed block H, then `last_block_height = H` and `last_block_app_hash = <hash returned by Commit for block H>`. If the app
-failed during the Commit of block H, then `last_block_height = H-1` and
-`last_block_app_hash = <hash returned by Commit for block H-1, which is the hash in the header of block H>`.
-
-We now distinguish three heights, and describe how CometBFT syncs itself with
-the app.
-
-```md
-storeBlockHeight = height of the last block CometBFT saw a commit for
-stateBlockHeight = height of the last block for which CometBFT completed all
-    block processing and saved all ABCI results to disk
-appBlockHeight = height of the last block for which ABCI app successfully
-    completed Commit
+last block it successfully completed `Commit` for. 
+
+The three steps performed before the state of a height is considered persisted are: 
+- The block is stored by CometBFT in the blockstore
+- CometBFT has stored the state (results returned from the application)
+- The application has committed it's state within `Commit`. 
+  
+The following diagram depicts the order in which these events happen, and the corresponding
+ABCI functions that are called and executed by CometBFT and the application:
+
+
+``` 
+APP:                                              Execute block                         Persist state
+                                                 /     return ResultFinalizeBlock            /
+                                                /                                           /  
+Event: ------------- block_stored -------------/ -------------state_stored ----------------/---app_persisted_state
+                          |                   /                   |                       /     |
+CometBFT: Decides - Persistes block ---Call `FinalizeBlock`  Persist results------------Commit---- 
+            on        in the                                (txResults, validator
+          Block      block store                                updated,..)
```

Comment:
```suggestion
``` 
APP:                                              Execute block                           Persist application
                                                / return ResultFinalizeBlock            / state
                                               /                                       /    |
Event: -------------- block_stored -----------/-------------state_stored -------------/---app_persisted_state
                          |                  /                   |                   /     
CometBFT: Decide ---- Persist ----------- Call ------------- Persist ------------ Call Commit---- 
          on Block    block in the        FinalizeBlock      results 
                      blockstore                             (txResults, validator updated,..)
```

Reply:
This is rendered weirdly :) I got the suggestion I think, but please confirm :)